interface/modules/custom_modules/oe-module-faxsms/library/webhook_receiver.php

```diff
@@ -0,0 +1,309 @@
+            [$status, json_encode($faxData), $faxSid, $siteId]
+        );
+        error_log("SignalWire Webhook: Updated existing fax {$faxSid} with status '{$status}' for site '{$siteId}'");
+    } else {
+        // Insert new fax
+        sqlStatement(
+            "INSERT INTO oe_faxsms_queue 
+             (job_id, calling_number, called_number, status, direction, details_json, site_id, date, receive_date) 
+             VALUES (?, ?, ?, ?, ?, ?, ?, NOW(), NOW())",
+            [
+                $faxSid,
+                $from,
+                $to,
+                $status,
+                $direction,
+                json_encode($faxData),
+                $siteId
+            ]
+        );
+        error_log("SignalWire Webhook: Inserted new fax {$faxSid} with status '{$status}' for site '{$siteId}'");
+    }
+
+    // If fax is received and has media, download and store it
+    if ($direction === 'inbound' && $status === 'received' && !empty($mediaUrl)) {
+        downloadAndStoreFaxMedia($faxSid, $mediaUrl, $from, $siteId, $existingFax['patient_id'] ?? 0);
+    }
+
+    // Respond with success
+    http_response_code(200);
+    header('Content-Type: application/xml');
+    echo '<?xml version="1.0" encoding="UTF-8"?><Response></Response>';
+    
+} catch (Exception $e) {
+    error_log("Error processing SignalWire webhook: " . $e->getMessage());
+    http_response_code(500);
+    exit('Internal server error');
+}
+
+/**
+ * Download fax media from SignalWire and store using FaxDocumentService
+ *
+ * @param string $faxSid
+ * @param string $mediaUrl
+ * @param string $fromNumber
+ * @param string $siteId
+ * @param int $patientId Patient ID if already assigned
+ * @return void
+ */
+function downloadAndStoreFaxMedia(
+    string $faxSid,
+    string $mediaUrl,
+    string $fromNumber,
+    string $siteId,
+    int $patientId = 0
+): void {
+    try {
+        // Get SignalWire credentials
+        $vendor = '_signalwire';
+        $credentials = sqlQuery(
+            "SELECT credentials FROM module_faxsms_credentials WHERE vendor = ? AND auth_user = 0",
+            [$vendor]
+        );
+
+        if (empty($credentials)) {
+            error_log("SignalWire Webhook: No credentials found for downloading fax media");
+            return;
+        }
+
+        $crypto = new CryptoGen();
+        $decrypted = $crypto->decryptStandard($credentials['credentials']);
+        $creds = json_decode($decrypted, true);
+
+        $projectId = $creds['project_id'] ?? '';
+        $apiToken = $creds['api_token'] ?? '';
+
+        if (empty($projectId) || empty($apiToken)) {
+            error_log("SignalWire Webhook: Invalid credentials for media download");
+            return;
+        }
+
+        // Download the fax media with authentication
+        // SignalWire files.signalwire.com requires Bearer token, not Basic auth
+        $ch = curl_init($mediaUrl);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
```

Comment:
Please use oeHttpRequest or Guzzle Http client, not curl. Using multiple different http clients leads to trouble and security vulnerabilities.